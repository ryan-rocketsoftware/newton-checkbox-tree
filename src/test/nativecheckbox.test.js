import React from 'react';
import { mount } from 'enzyme';
import { assert } from 'chai';

import NativeCheckbox from '../js/NativeCheckbox';

describe('<NativeCheckbox />', () => {
    describe('snapshot', () => {
        it('should render NativeCheckbox component correctly', () => {
            const NativeCheckboxComponent = mount(<NativeCheckbox />);
            expect(NativeCheckboxComponent).toMatchSnapshot();
        });
    })
    describe('indeterminate', () => {
        it('should set the JavaScript property to true when true', () => {
            const wrapper = mount(
                <NativeCheckbox indeterminate />,
            );

            assert.isTrue(wrapper.find('input').getDOMNode().indeterminate);
        });

        it('should set the JavaScript property to false when not true', () => {
            const wrapper = mount(
                <NativeCheckbox />,
            );

            assert.isFalse(wrapper.find('input').getDOMNode().indeterminate);
        });
    });
});
