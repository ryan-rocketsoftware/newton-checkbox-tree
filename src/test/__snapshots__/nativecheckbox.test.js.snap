// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NativeCheckbox /> snapshot should render NativeCheckbox component correctly 1`] = `ReactWrapper {}`;
